### PR TITLE
Fix issues with user-defined `hx` package

### PR DESCRIPTION
**Pull request by RealyUniqueName**
See HaxeFoundation/haxe#8543 for the issue and HaxeFoundation/haxe#8561 for PR
There are more `hx::` references to fix in cpp/h files probably.

### CODE REVIEW

**Comment by RealyUniqueName** [comment]
CI failure is not related. Same thing happens in master.

**Comment by Simn** [comment]
@hughsando Could you check this and the linked Haxe PR?

**Comment by hughsando** [comment]
The Macro.h file is generated from the Macro.tpl and MacrosFixed.h, so should not be edited directly. Due to some legacy reasons, '<::hx' should be written as '< ::hx' (nb: extra space).

**Comment by RealyUniqueName** [comment]
updated

File: include/hx/MacrosFixed.h [hunk 1]
```diff
@@ -81,42 +81,42 @@ static  ::Dynamic Create##enum_obj(::String inName,hx::DynamicArray inArgs) \
...
+   ::hx::ObjectPtr< ::hx::Class_obj > __GetClass() const { return __mClass; } \
+   inline static ::hx::ObjectPtr< ::hx::Class_obj> &__SGetClass() { return __mClass; } \
+   inline operator super *() { return this; }
 
 #define HX_DO_RTTI \
    HX_DO_RTTI_ALL \
-    ::hx::Val __Field(const ::String &inString, hx::PropertyAccess inCallProp); \
-    ::hx::Val __SetField(const ::String &inString,const  ::hx::Val &inValue, hx::PropertyAccess inCallProp); \
+    ::::hx::Val __Field(const ::String &inString, ::hx::PropertyAccess inCallProp); \
+    ::::hx::Val __SetField(const ::String &inString,const  ::::hx::Val &inValue, ::hx::PropertyAccess inCallProp); \
```
Comment (Aurel300) [inline include/hx/MacrosFixed.h]: These seem wrong?

File: include/hx/MacrosFixed.h [hunk 2]
```diff
@@ -81,42 +81,42 @@ static  ::Dynamic Create##enum_obj(::String inName,hx::DynamicArray inArgs) \
...
-   static hx::ObjectPtr<hx::Class_obj> __mClass; \
-   static hx::ObjectPtr<hx::Class_obj> &__SGetClass() { return __mClass; } \
+   static ::hx::ObjectPtr< ::hx::Class_obj> __mClass; \
+   static ::hx::ObjectPtr< ::hx::Class_obj> &__SGetClass() { return __mClass; } \
 	static void __register();
 
 #define HX_DO_ENUM_RTTI_INTERNAL \
    HX_DO_RTTI_BASE  \
-    ::hx::Val __Field(const ::String &inString, hx::PropertyAccess inCallProp); \
+    ::::hx::Val __Field(const ::String &inString, ::hx::PropertyAccess inCallProp); \
```
Comment (Aurel300) [inline include/hx/MacrosFixed.h]: Here also.

**Comment by Simn** [comment]
CI is failing, PR needs an update.

**Pull request merged**